### PR TITLE
Allow variable overriding by remove variables.scss

### PR DESCRIPTION
**Pull request by msecret**
This allows users of the library to override global variables like color
by importing their variables before the USWDS library. This ensures only
defaulted vars are imported, meaning if the value is already set, such
as by the user, then it won't be overriden. This also ensures the USWDS
components will receive the new colors.

More information can be found here:
https://github.com/18F/web-design-standards/issues/952

Closes 952.

### CODE REVIEW

**Comment by maya** [comment]
@msecret we're using release branches. Can you open this PR against `v0.9.x`?

**Comment by msecret** [comment]
@maya can you update the dev instructions then: All development and pull requests should be done against the 18f-pages-staging branch.

**Comment by maya** [comment]
@msecret re: dev instructions - sounds like we may change our git workflow so will leave it as is until we discuss next week.

Also, @rossdakin had some feedback in #952.

**Comment by maya** [comment]
@msecret please update the docs that refer to `variables.scss`

**Comment by maya** [comment]
If we want to get this into the next release, this PR needs to:
- [ ] update change log
- [ ] re-make the ZIP file

**Comment by msecret** [comment]
Are we doing any releases before 1.0?

File: package.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
   "description": "Open source UI components and visual style guide for U.S. government websites",
   "main": "src/js/start.js",
   "scripts": {
-    "prepublish": "gulp no-gem build"
```
Comment (maya) [inline package.json]: Does this still need to be removed or was it meant to be temporary?

File: docs/pages/getting-started.html [hunk 2]
```diff
@@ -21,8 +21,10 @@ <h3 class="usa-heading">For developers</h3>
   <li>Uses a <strong><a href="https://pages.18f.gov/frontend/css-coding-styleguide/naming/">modified BEM</a></strong> approach created by 18F for naming CSS selectors. Objects in CSS are separated by single dashes. Multi-word objects are separated by an underscore (For example: <code>.usa-button-cool_feature-active</code>).</li>
   <li>Uses <strong>modular CSS</strong> for scalable, modular, and flexible code.</li>
   <li>Uses <strong>nesting</strong> when appropriate. Nest minimally with up to two levels of nesting.</li>
-  <li><strong>Global variables</strong> are defined in the _variables.scss file. This is where custom theming can be done to change site-wide colors, fonts, etc.</li>
-  <li>Hard-coded magic numbers are avoided and, if necessary, defined in the <code>_variables.scss</code> file.</li>
+  <li><strong>Global variables</strong> are defined in the <code>_defaults.scss</code> file.
+    Any of these variables can be changed by a user of the library by
+    overwriting them with their own Sass file.</li>
```
Comment (maya) [inline docs/pages/getting-started.html]: "overwriting" should be "overriding".

File: docs/pages/getting-started.html [hunk 3]
```diff
@@ -21,8 +21,10 @@ <h3 class="usa-heading">For developers</h3>
   <li>Uses a <strong><a href="https://pages.18f.gov/frontend/css-coding-styleguide/naming/">modified BEM</a></strong> approach created by 18F for naming CSS selectors. Objects in CSS are separated by single dashes. Multi-word objects are separated by an underscore (For example: <code>.usa-button-cool_feature-active</code>).</li>
   <li>Uses <strong>modular CSS</strong> for scalable, modular, and flexible code.</li>
   <li>Uses <strong>nesting</strong> when appropriate. Nest minimally with up to two levels of nesting.</li>
-  <li><strong>Global variables</strong> are defined in the _variables.scss file. This is where custom theming can be done to change site-wide colors, fonts, etc.</li>
-  <li>Hard-coded magic numbers are avoided and, if necessary, defined in the <code>_variables.scss</code> file.</li>
+  <li><strong>Global variables</strong> are defined in the <code>_defaults.scss</code> file.
+    Any of these variables can be changed by a user of the library by
```
Comment (maya) [inline docs/pages/getting-started.html]: We aren't really changing the variables themselves, but the variable values.

**Comment by maya** [comment]
I see some discrepancies between defaults and variables and fixed them in: https://github.com/18F/web-design-standards/pull/1125

File: docs/_visual/typography.md [hunk 4]
```diff
@@ -930,7 +930,8 @@ order: 01
     <h4 class="usa-heading">Implementation</h4>
     <p>To get the max-width on body text, add the class <code>usa-content</code> to your document. Use at the specificity that best suits your project's needs.</p>
     <p>Lists must use <code>usa-content-list</code> for the above.</p>
-    <p>You can change the max-width value <code>$text-max-width</code> in <code>assets/_scss/core/<wbr>variables.scss</code>.</p>
+    <p>You can change the max-width value <code>$text-max-width</code> by
+    overwriting the variables in <code><wbr>_defaults.scss</code>.</p>
```
Comment (maya) [inline docs/_visual/typography.md]: Overwriting should be overriding. 

Need to tell them they should do this in their own custom Sass files and not directly in `_defaults.scss`.

File: docs/pages/getting-started.html [hunk 5]
```diff
@@ -21,8 +21,10 @@ <h3 class="usa-heading">For developers</h3>
   <li>Uses a <strong><a href="https://pages.18f.gov/frontend/css-coding-styleguide/naming/">modified BEM</a></strong> approach created by 18F for naming CSS selectors. Objects in CSS are separated by single dashes. Multi-word objects are separated by an underscore (For example: <code>.usa-button-cool_feature-active</code>).</li>
   <li>Uses <strong>modular CSS</strong> for scalable, modular, and flexible code.</li>
   <li>Uses <strong>nesting</strong> when appropriate. Nest minimally with up to two levels of nesting.</li>
-  <li><strong>Global variables</strong> are defined in the _variables.scss file. This is where custom theming can be done to change site-wide colors, fonts, etc.</li>
-  <li>Hard-coded magic numbers are avoided and, if necessary, defined in the <code>_variables.scss</code> file.</li>
+  <li><strong>Global variables</strong> are defined in the <code>_defaults.scss</code> file.
+    Any of these variable's values can be changed by a user of the library by
+    overriding them with their own Sass file.</li>
```
Comment (maya) [inline docs/pages/getting-started.html]: Overwriting should be overriding.

**Comment by juliaelman** [comment]
Placing this into the backlog until rebased.

**Comment by juliaelman** [comment]
Closing this in favor of the solve in PR #1098

**Comment by maya** [comment]
I think this is still valid - that we don't need a variables file in addition to the defaults file. 

cc @msecret

**Comment by juliaelman** [comment]
@maya please feel free to review #1098 and add any feedback. Discussed with Marco and the goal of those two pull requests are pretty similar. If you decide to review it, please feel free to assign yourself to that pull request - thank you! <3

**Comment by maya** [comment]
Sure, I'm thinking more around the variables file being duplicative, we can add an issue around that.

**Pull request closed without merging**